packages/frontend-2/components/settings/workspaces/members/Table.vue

```diff
@@ -0,0 +1,181 @@
+      { id: 'verified', header: 'Status', classes: 'col-span-3' },
+      { id: 'role', header: 'Role', classes: 'col-span-2' }
+    ]"
+    :items="members"
+  >
+    <template #name="{ item }">
+      <div class="flex items-center gap-2">
+        <UserAvatar :user="item" />
+        <span class="truncate text-body-xs text-foreground">{{ item.name }}</span>
+      </div>
+    </template>
+    <template #company="{ item }">
+      <span class="text-body-xs text-foreground">
+        {{ item.company ? item.company : '-' }}
+      </span>
+    </template>
+    <template #verified="{ item }">
+      <span class="text-body-xs text-foreground-2">
+        {{ item.verified ? 'Verified' : 'Unverified' }}
+      </span>
+    </template>
+    <template #role="{ item }">
+      <FormSelectWorkspaceRoles
+        :model-value="item.role as WorkspaceRoles"
+        fully-control-value
+        :disabled="!isCurrentUser(item.id)"
+        @update:model-value="
+          (newRoleValue) => openChangeUserRoleDialog(item, newRoleValue)
+        "
+      />
+    </template>
+  </LayoutTable>
+
+  <SettingsSharedChangeRoleDialog
+    v-model:open="showChangeUserRoleDialog"
+    :name="userToModify?.name ?? ''"
+    :old-role="oldRole"
+    :new-role="newRole"
+    @update-role="onUpdateRole"
+  />
+</template>
+
+<script setup lang="ts">
+// Todo: Enable searching once supported
+import type { WorkspaceRoles } from '@speckle/shared'
+// import { MagnifyingGlassIcon } from '@heroicons/vue/24/outline'
+// import { useDebouncedTextInput } from '@speckle/ui-components'
+import { settingsWorkspacesMembersQuery } from '~~/lib/settings/graphql/queries'
+import { workspaceUpdateRoleMutation } from '~~/lib/workspaces/graphql/mutations'
+import { useActiveUser } from '~~/lib/auth/composables/activeUser'
+import { useMutation, useQuery } from '@vue/apollo-composable'
+import { useGlobalToast, ToastNotificationType } from '~~/lib/common/composables/toast'
+import {
+  convertThrowIntoFetchResult,
+  getCacheId,
+  getFirstErrorMessage
+} from '~~/lib/common/helpers/graphql'
+import type { SettingsWorkspacesMembersQuery } from '~~/lib/common/generated/gql/graphql'
+
+type UserItem = {
+  id: string
+  role: string
+  name: string
+  verified?: boolean | null
+  avatar?: string | null
+  company?: string | null
+}
+
+const props = defineProps<{
+  workspaceId: string
+}>()
+
+const { triggerNotification } = useGlobalToast()
+// const { on, bind, value: search } = useDebouncedTextInput()
+const { activeUser } = useActiveUser()
+const { mutate: updateChangeRole } = useMutation(workspaceUpdateRoleMutation)
+
+const { result } = useQuery<SettingsWorkspacesMembersQuery>(
+  settingsWorkspacesMembersQuery,
+  () => ({
+    workspaceId: props.workspaceId
+  })
+)
+
+const showChangeUserRoleDialog = ref(false)
+const newRole = ref<WorkspaceRoles>()
+const userToModify = ref<UserItem>()
+
+const members = computed(() => {
```

Comment:
can be shortened
```ts
const members = computed(() => (result.value?.workspace.team || []).map(({ user, ...rest }) => ({...user, ...rest })))
```

Reply:
Also, there's a few unnecessary optional chaining operators, only add `?` operators where needed (TS will tell you)

`result?` - result will never be falsy, its a ref
`result.value.workspace?` - will never be falsy, the return type in the GQL schema is non-null, the resolver throws if a workspace isn't found (which should probably be handled somehow too)